server/resp/src/main/java/org/infinispan/server/resp/commands/connection/AUTH.java

```diff
@@ -23,7 +24,11 @@ public AUTH() {
    public CompletionStage<RespRequestHandler> perform(Resp3AuthHandler handler,
                                                       ChannelHandlerContext ctx,
                                                       List<byte[]> arguments) {
-      CompletionStage<Boolean> successStage = handler.performAuth(ctx, arguments.get(0), arguments.get(1));
+      CompletionStage<Boolean> successStage = handler.performAuth(ctx, arguments.get(0), arguments.get(1))
+            .thenApply(s -> {
+               if (s) Consumers.OK_BICONSUMER.accept(null, handler.allocator());
```

Comment:
The allocator should only be used in the callback from stageToReturn or without a lambda.